### ISSUE TITLE
wiki: CodeReviewComments change - Justify variable name length

### ISSUE THREAD

**Issue by stevenpitts** [golang/go#57181]
From https://github.com/golang/go/wiki/CodeReviewComments#variable-names:

> Variable names in Go should be short rather than long.  This is especially true for local variables with limited scope.  Prefer `c` to `lineCount`.  Prefer `i` to `sliceIndex`.
> The basic rule: the further from its declaration that a name is used, the more descriptive the name must be. For a method receiver, one or two letters is sufficient. Common variables such as loop indices and readers can be a single letter (`i`, `r`). More unusual things and global variables need more descriptive names.

Similarly to https://github.com/golang/go/issues/38912, I don't understand why this is suggested. The comments there have some explanations I don't agree with, but they also suggest that I'm free to do things my own way if I disagree with the document. Literally this is of course true, but the spirit of it conflicts with the introduction of the page:

> This is a laundry list of common mistakes, not a comprehensive style guide.

It says that violating this would be a *mistake*. That's a pretty strong word, so the section on variable names can easily be (mis?)interpreted as [canonical](https://google.github.io/styleguide/go/#canonical).

I'll assume that the introduction is the better source of truth. Including justification would also suit the described use case of the document:

> so that a single detailed explanation can be referred to by shorthands

If someone's PR gets a comment with a link to this section, I imagine a detailed explanation would mean a less divisive resolution.

Could the variable naming convention section be enhanced to include justification?

**Comment by ianlancetaylor** [comment]
In particular see https://google.github.io/styleguide/go/decisions#variable-names.

I added a couple of links to the CodeReviewComments wiki page.  Ultimately this is a style decision, and as such justification is not required.  The goal is not to provide arguments in favor of all style decisions.  Sometimes there are several good choices, and you just have to make a choice.

And the CodeReviewComments wiki page really is optional.  I changed "mistakes" to "style issues".  If you don't want to follow this particular style, that's fine.
